### PR TITLE
SNOW-364955: bug fix in pyobject raii class

### PR DESCRIPTION
**Pull request by sfc-gh-wshangguan**
SNOW-364955

1. Fix a potential bug in RAII class, which doesn't correctly release the original object.
2. Clean some code.

### CODE REVIEW

**Comment by sfc-gh-mkeller** [comment]
Hey @sfc-gh-wshangguan 
I can't see the bug that you are fixing. Could you please point it out?
This looks like a code cleanup to me

**Comment by sfc-gh-wshangguan** [comment]
> Hey @sfc-gh-wshangguan
> I can't see the bug that you are fixing. Could you please point it out?
> This looks like a code cleanup to me

Sure man, I planned to tag you after I passed all tests. But let me point it out now

File: src/snowflake/connector/cpp/ArrowIterator/Python/Common.hpp [hunk 1]
```diff
@@ -35,13 +35,13 @@ class UniqueRef
...
 
-  UniqueRef(UniqueRef&& other) : UniqueRef(other.detach())
+  explicit UniqueRef(UniqueRef&& other) : UniqueRef(other.release())
   {
   }
 
   UniqueRef& operator=(UniqueRef&& other)
   {
-    m_pyObj = other.detach();
+    reset(other.release());
```
Comment (sfc-gh-wshangguan) [inline src/snowflake/connector/cpp/ArrowIterator/Python/Common.hpp:44]: @sfc-gh-mkeller  this is what I think a potential bug. Looks like the `m_pyObj` was not properly released by `Py_XDECREF`

Comment (sfc-gh-mkeller) [reply src/snowflake/connector/cpp/ArrowIterator/Python/Common.hpp:44]: Ahh, I see. Makes sense. Thank you!

**Review by sfc-gh-mkeller — APPROVED** [review]
🚢 🚢

**Pull request merged**